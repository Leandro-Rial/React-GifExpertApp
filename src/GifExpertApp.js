import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
           <h2>GifExpertApp</h2>
           <AddCategory setCategories={setCategories} />
           <hr /> 
           <ol>
                {
                    categories.map(( category, i ) => (
                        <GifGrid category={category} key={i} />
                    ))
                }
           </ol>
        </>
    )
}

export default GifExpertApp
