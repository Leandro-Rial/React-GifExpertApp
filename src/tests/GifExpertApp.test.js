import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas GifExpertApp', () => {
    
    test('debe de mostar <GifExpertApp /> correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp />)
        
        expect(wrapper).toMatchSnapshot()
        
    });
    
    test('debe de mostrar una lista de categories', () => {
        
        const categories = ["Dragon Ball", "One Punch"]
        
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find("GifGrid").length).toBe(categories.length)

    });

});