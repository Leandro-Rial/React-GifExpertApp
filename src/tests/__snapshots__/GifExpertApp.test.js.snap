// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifExpertApp debe de mostar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas GifExpertApp debe de mostrar una lista de categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="0"
    />
    <GifGrid
      category="One Punch"
      key="1"
    />
  </ol>
</Fragment>
`;
